lib/datasets.go

```diff
@@ -975,6 +975,7 @@ func (datasetImpl) Save(scope scope, p *SaveParams) (*dataset.Dataset, error) {
 			if err := scope.Logbook().RemoveLog(ctx, ref); err != nil {
 				log.Errorf("couldn't cleanup unused reference: %q", err)
 			}
+			go scope.sendEvent(event.ETDatasetCreateFail, ref.InitID, ref.InitID)
```

Comment:
Is there a reason this needs to be a separate type of event? It appears that this does the same thing as `ETDatasetDeleteAll`, could we just reuse it here?

Reply:
Yea.... I'm of two minds on this. This pr shows which side I ended up falling on. On one hand, adding an additional event for a course of action that is so similar (and currently has identical consequences) to an action we already have defined seems like it has the potential for bugs. On the other hand, the events talk about two different circumstances, in one case we are alerting "hey the user said remove this entire dataset", in the other "hey we failed creating a dataset", and that might actually have different consequences in the future (although, tbh I can't think of any rn).

I've been thinking of events as a way of the code path saying "hey this is an important moment, thought you should know it happened", rather than "I know I need you to do this thing and this is the event that tells you to do it", so I leaned toward making a new event that describes the circumstance. That could also be the totally wrong lens!

Reply:
> although, tbh I can't think of any rn

There's your answer. If you don't know of a way things _will_ be separate in the very near future, avoid introducing new code. We should be policing the list of event types _very_ hard, and have a small number of well-defined event types that re-use wherever appropriate. This is a good example.

In a situation like this the important thing to do is to read the comment describing the event you're about to overlay onto, and check if your use case fits within the description. If not, modify the description and/or the event name itself until it accurately describes _both_ your use case & the existing uses.